### PR TITLE
Handle add_languages(native:) in introspector

### PR DESCRIPTION
**Pull request by jon-turney**
This should have been part of #6532

### CODE REVIEW

**Comment by mensinda** [comment]
This should have a unit test if possible (just search for `introspect` in `run_unittests.py` for templates).

**Comment by jpakkane** [comment]
Ping.

**Comment by jon-turney** [comment]
> This should have a unit test if possible

Reading https://github.com/mesonbuild/meson/pull/6512#issuecomment-599989205, I think the test needed here is to verify that e.g. c_args/build.c_args options are present in introspection output appropriately.

File: mesonbuild/ast/introspection.py [hunk 1]
```diff
@@ -137,17 +138,24 @@ def do_subproject(self, dirname: str) -> None:
...
-                if isinstance(lang, StringNode):
-                    assert isinstance(lang.value, str)
-                    lang = lang.value
-                if not isinstance(lang, str):
-                    continue
-                lang = lang.lower()
-                if lang not in self.coredata.compilers[for_machine]:
-                    self.environment.detect_compiler_for(lang, for_machine)
+        if 'native' in kwargs:
+            native = kwargs.get('native', False)
```
Comment (mensinda) [inline mesonbuild/ast/introspection.py:142]: You should `flatten_kwargs` the `kwargs` first.

Comment (jon-turney) [reply mesonbuild/ast/introspection.py:143]: Done

**Review by mensinda — APPROVED** [review]
LGTM once the CI is green

File: mesonbuild/ast/introspection.py (hunk 1 above)
Comment (jon-turney) [reply mesonbuild/ast/introspection.py:143]: ... and it looks like it was never working correctly due to that, so I had the expected test result upside-down?

**Review by Ericson2314 — APPROVED** [review]

**Pull request merged**